### PR TITLE
Add redis support in backend

### PR DESCRIPTION
**Pull request by Flova**

### CODE REVIEW

File: lecture2gether_flask/app.py [hunk 1]
```diff
@@ -96,19 +112,23 @@ def on_join(data):
...
+            db.hset('rooms', room_token, json.dumps(room))
 
-    if not room_token in ROOM_STATES.keys():
+            join_room(room_token)
+        
+        # Send response anyway
+        emit('video_state_update', room['state'], room=request.sid)
+        send({'status_code': 200}, room=request.sid)
+        
+    else:  # Room does not exist
```
Comment (Flova) [inline lecture2gether_flask/app.py:128]: Move stuff up for less indentation

File: lecture2gether_flask/app.py [hunk 2]
```diff
@@ -118,46 +138,55 @@ def on_leave(data):
...
+        send({'status_code': 404}, room=request.sid)
+        return
 
 @socketio.on('video_state_set')
 def on_video_state_set(state):
     """Update a watch room"""
-    global ROOM_STATES
-
     if 'roomId' not in state:
         state = {'status_code': 400}
```
Comment (Flova) [inline lecture2gether_flask/app.py:165]: This could be done inline

Comment (JanGut) [reply lecture2gether_flask/app.py]: What?

File: lecture2gether_flask/app.py (hunk 1 above)
Comment (JanGut) [reply lecture2gether_flask/app.py]: Done.

File: lecture2gether_flask/app.py [hunk 3]
```diff
@@ -8,22 +8,37 @@
...
 from flask import Flask, request, jsonify, abort
-from flask_socketio import SocketIO, join_room, leave_room, close_room, send, emit, rooms
 from flask_cors import CORS
+from flask_socketio import SocketIO, join_room, leave_room, close_room, send, emit, rooms
+from redis.client import Redis
 
-app = Flask(__name__)
 
+logging.basicConfig(level=os.getenv('LOGLEVEL', 'INFO'))
+# TODO: logger log file
```
Comment (JanGut) [inline lecture2gether_flask/app.py:17]:
```suggestion
```

File: lecture2gether_flask/app.py [hunk 4]
```diff
@@ -8,22 +8,37 @@
...
+# TODO: readme redis
```
Comment (JanGut) [inline lecture2gether_flask/app.py:18]:
```suggestion
```
Link install guide in readme

**Pull request merged**